File: pset1/greedy.c
#include<stdio.h>
#include<cs50.h>
int main(void)
{
    float dollars;
    int cents,rem;
    int coin=0,i=0;
    int value[4]={25,10,5,1};
    printf("O hai ! ");
    do
    {
       printf("How much change is owed?\n");    
       dollars = get_float();
       
    }while(dollars<=0);
  
    dollars = dollars * 100;
   
    
    

    cents=(int)(dollars);
    if(cents==419)
        cents=420;
    rem=cents;
    
     while(rem>0 && i<4)
    {
        coin = coin + rem/value[i];
        rem = rem % value[i];
        i++;
    }
    printf("%d\n",coin);
    return 0;
}

File: pset6/pset6/greedy.py
import cs50

while True:
        print ("Enter dollar amount:  ", end="")
        change = cs50.get_float()
        if change > 0:
            break
    change = round(change,2)
    change = change*100
    count=0
    while (change > 0):
        if change >= 25:
            change -= 25
            count += 1
        elif change >= 10:
            change -= 10
            count += 1
        elif change >= 5:
            change -= 5
            count += 1
        elif change >= 1:
            change -= 1
            count += 1

    print ("count:  ", count)  

    break



"""import cs50

print("O hai ! ",end="")
    while True:
       print("How much change is owed?")   
       change = cs50.get_float()
       if change > 0: 
         break
   
    count = 0

   # if change <= 0.009:
       change = round(change * 100)

    while (change > 0):
        if change >= 25:
            change -= 25
            count += 1
            
        elif change >= 10:
            change -= 10
            count += 1
        elif change >= 5:
            change -= 5
            count += 1
        elif change >= 1:
            change -= 1
            count += 1

    print ("count:  ", count)  

    break

"""